### ISSUE TITLE
Improve using `.target_number` in Story Table

### ISSUE THREAD

**Issue by plocket** [plocket/docassemble-cucumber#706]
- [x] Allow `.target_number` of 0 to set `.there_are_any` to `False`.
- [x] Allow `.target_number` to set `.there_are_any` in general so the author doesn't need to set `.there_are_any` separately.
- [x] Fix `.target_number` using one fewer loop than it should.
- [x] Create feature tests for those items.

**Comment by plocket** [comment]
This is a wilder ride than I anticipated. Adding some record of the digging so far. This is how the story table flow currently works for the relevant code:

Notes: Each 'row' is an object representing a row.

- for each page
  - for each `target_number` row
    - keep the `target_number` row (don't do anything to it)
    - create the `there_are_any` row with a reference to the original `target_number` as its `source` property
      - get the `there_are_any` `value`
    - create `there_is_another` row with a reference to the original `target_number` as its `source` property
      - get the `there_is_another` `value` depending on the accumulated `times_used` value in the original `target_number` row

  - for each field
    - find which table row matches the field's variable name
    - set the answer of the field using the table row value
    - increment `times_used` of the row and also of the `source` object in the row. Mutating the `source` is what allows `target_number` to persistently accumulate the times it's used and later tell `there_is_another` which value it should for that particular page.

Problem: Take the following row

```
| children.target_number | 2 |  |
```

If `there_are_any` needs to be set by the user, it adds to the number of times `target_number` is used (`times_used`). That counts 1 use for `there_are_any` and 1 use for `there_is_another`, coming out to the correct 2 uses. But if the author sets `there_are_any` themselves, that first accumulation doesn't happen. The test only counts 1 use for `there_is_another` since `there_are_any` never gets asked. In that case, the test thinks it hasn't added enough children and tries to add another child.

My solution: We need to track `there_are_any` separately from `there_is_another` for `target_number` rows. Then we use only the `there_is_another` accumulator to get a consistent number. We can't use that number to determine (for the report) which rows were used through the whole test. Example:

```
| children.target_number | 0 |  |
```

That should correctly set `.there_are_any` to `False` and will be a legitimate use of the `target_number` row, but will not show up in the `there_is_another` accumulator.

In order to write a correct report of what rows were used, we can either keep `times_used` as it is or add up the two separate accumulators and get rid of `times_used`.

**Comment by plocket** [comment]
Closed by #709
